### PR TITLE
Update to substrate v3.0.0

### PR DESCRIPTION
**Pull request by clearloop**
## DESC

Merged #53 by default, since the(/our) wasmi package also has a lot of changes after the last commit of this repo

### CODE REVIEW

**Comment by clearloop** [comment]
```
Error: Input("Other: Essential task failed.")
```

After https://github.com/paritytech/substrate/pull/8155

File: client/service/src/task_manager/mod.rs [hunk 1]
```diff
@@ -154,15 +155,28 @@ impl SpawnEssentialTaskHandle {
...
 			.catch_unwind()
-			.map(move |_| {
-				log::error!("Essential task `{}` failed. Shutting down service.", name);
-				let _ = essential_failed.close_channel();
+			.map(move |r| {
+				// Workaround
+				if r.is_err() {
+					log::error!("Essential task `{}` failed. Shutting down service.", name);
+					let _ = essential_failed.close_channel();
+				}
```
Comment (clearloop) [inline client/service/src/task_manager/mod.rs:163]: Workaround

File: client/service/src/task_manager/mod.rs [hunk 2]
```diff
@@ -154,15 +155,28 @@ impl SpawnEssentialTaskHandle {
...
 			});
```
Comment (clearloop) [inline client/service/src/task_manager/mod.rs:164]: Workaround

Comment (clearloop) [reply client/service/src/task_manager/mod.rs:164]: https://github.com/paritytech/substrate/blob/3c1c49e8317abfedaede198ef4aed8173eb23ef0/client/service/src/task_manager/mod.rs#L189

Comment (clearloop) [reply client/service/src/task_manager/mod.rs:164]: https://doc.rust-lang.org/std/panic/struct.AssertUnwindSafe.html

**Review by atenjin — APPROVED** [review]

**Review by atenjin — APPROVED** [review]

**Pull request merged**